packages/nuxt/src/app/utils.ts

```diff
@@ -1,4 +1,49 @@
-/** @since 3.9.0 */
+import { useNuxtApp } from './nuxt'
+
 export function toArray<T> (value: T | T[]): T[] {
   return Array.isArray(value) ? value : [value]
 }
+
+type CallbackFn = () => void
+type ObserveFn = (element: Element, callback: CallbackFn) => () => void
+
+export function useObserver (): { observe: ObserveFn } | undefined {
```

Comment:
I'd rename this to useIntersectionObserver, maybe it's something Nuxt can export in the future.

Reply:
Sure thing! I'll rename it.

Reply:
Renamed to useIntersectionObserver